modules/misunderstood/src/views/full/ApiClient.ts

```diff
@@ -0,0 +1,57 @@
+import { AxiosRequestConfig, AxiosStatic } from 'axios'
+
+import { FLAGGED_MESSAGE_STATUS, RESOLUTION_TYPE, ResolutionData } from '../../types'
+
+const MODULE_URL_PREFIX = '/mod/misunderstood'
+
+class ApiClient {
+  constructor(private axios: AxiosStatic) {}
+
+  async get(url: string, config?: AxiosRequestConfig) {
+    const res = await this.axios.get(url, config)
+    return res.data
+  }
+
+  async post(url: string, data?: any, config?: AxiosRequestConfig) {
+    const res = await this.axios.post(url, data, config)
+    return res.data
+  }
+
+  getForModule(url: string, config?: AxiosRequestConfig) {
+    return this.get(MODULE_URL_PREFIX + url, config)
+  }
+
+  postForModule(url: string, data?: any, config?: AxiosRequestConfig) {
+    return this.post(MODULE_URL_PREFIX + url, data, config)
+  }
+
+  getEventCounts(language: string) {
+    return this.getForModule('/events/count', {
+      params: {
+        language
+      }
+    })
+  }
+
+  getEvents(language: string, status: string) {
+    return this.getForModule(`/events/${status}`, {
+      params: {
+        language
+      }
+    })
+  }
+
+  getEvent(id: string) {
+    return this.getForModule(`/events/${id}`)
+  }
+
+  updateStatus(id: string, status: FLAGGED_MESSAGE_STATUS, resolutionData?: ResolutionData) {
+    return this.postForModule(`/events/${id}/status`, { status, ...resolutionData })
+  }
+
+  applyAllPending() {
+    return this.postForModule(`/apply-all-pending`)
+  }
+}
+
+export default ApiClient
```

Comment:
What do you think of using a similar logic to ours for the view API ? It would be easier to maintain

https://github.com/botpress/botpress/blob/master/modules/nlu/src/views/api.ts

https://github.com/botpress/botpress/blob/patch/12-1-6/modules/hitl/src/views/full/api.ts

Reply:
I would be thankful for the specific pointers how it can be made more similar. So far I don't see it much different other than having 4 util methods to keep the code DRYer and the usage of a class instead of an object factory. Can convert it to factory of course but the class has the advantage of getting its interface automatically.